### ISSUE TITLE
Bug: ( no-shallow throw error for two names witch are not exactly the same  )

### ISSUE THREAD

**Issue by morijenab** [eslint/eslint#15075]
### Environment

Node version: 14.17.5
npm version: 6.14.14
Local ESLint version: 7.32
Global ESLint version: 
Operating System: win 10

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```
"no-shadow": "error"
```
</details>

 `no-shallow` throw error for these declarations which don't have the same name 
```js
export enum SymbolState {
}

export interface Symbol {
}
```

### What did you expect to happen?

if a part of a name is repeated in the other declaration name it should be considered fine as far as they are not exactly the same.

### What actually happened?

 `no-shallow`  throw error for two names which are not exactly the same

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by aladdin-add** [comment]
if using typescript, you'll be able to use [@typescript-eslint/no-shadow](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-shadow.md).
